teaching-platfrom/db_handler.py

```diff
@@ -0,0 +1,98 @@
+import yaml
+import psycopg2
+import logging
+from flask import Flask
+import jsonpickle
+from enum import Enum
+from flask import request
+
+with open("../config.yaml", "r") as f:
+    config = yaml.load(f, Loader=yaml.FullLoader)
+logging.basicConfig(level=logging.INFO,filemode="w", filename="../logs.log")
+jsonpickle.set_preferred_backend('json')
+jsonpickle.set_encoder_options('json', ensure_ascii=False)
+
+def get_connection_to_db():
+    db_config = config["database"]
+    conn = psycopg2.connect(host=db_config["host"],dbname = db_config["dbname"],
+                            user = db_config["user"], port = db_config["port"], password = db_config["password"]
+    )
+    return conn
+
+class User:
+    def __init__(self, id, username, password="-"):
+        self.id = id
+        self.username = username
+        self.password = password
+
+class CensuredUser:
+    def __init__(self, id, username):
+        self.id = id
+        self.username = username
+
+class Role(Enum):
+    STUDENT = 0
+    TEACHER = 1
+    ADMIN = 2
+
+def add_user(username, password):
+    conn = get_connection_to_db()
+    cur = conn.cursor()
+
+    cur.execute("SELECT username FROM users")
+    results = cur.fetchall()
+    if (username,) in results:
+        logging.warning("Username alreday taken, pick a diffrent one")
+        conn.close()
+        return 409
+
+    query = f"INSERT INTO users (username, password,role) VALUES ('{username}', '{password}','USER')"
+    cur.execute(query)
+    conn.commit()
+    logging.info(f"User {username} added")
+
+    conn.close()
+    return 200
```

Comment:
`return 200` / `return 400` (or any other status) from the point of view of `db_handler` says very little. Maybe consider applying some sort of common interface between those two components (the REST service and the db handler).
A good approach would be proper exception handling.
For example - you can create `UserAlreadyExists` exception class with status code and message as attributes. `db_handler` can throw this exception and `controller` can catch it and handle it accordingly.

It's an [EAFP](https://docs.python.org/3/glossary.html#term-EAFP) approach, so not everybody loves it but you can discuss it with @mstopa.

Reply:
Done.